libvast/src/format/mrt.cpp

```diff
@@ -87,12 +120,11 @@ expected<event> reader::read() {
   mrt::record r;
   if (!parser_(buffer_, r))
     return make_error(ec::format_error, "failed to parse MRT message");
+  if(r.header.subtype == 1)
+    VAST_DEBUG(this->name(), "type", r.header.type, "subtype", r.header.subtype);
   // Convert
-  auto e = event{visit(factory{}, r.message)};
   // Take the timestamp from the Common Header as event time.
-  std::chrono::duration<uint32_t> since_epoch{r.header.timestamp};
-  auto ts = timestamp{std::chrono::duration_cast<timespan>(since_epoch)};
-  e.timestamp(ts);
+  visit(factory{events_, r.header.timestamp}, r.message);
```

Comment:
Why not pass the `vast::timestamp` into the factory constructor? We have to perform the conversion from `uint32_t` anyway, and right here is a good position. This way, the factory has exactly one purpose, which is always a more flexible design.